### PR TITLE
Upadate xml_parser autogenerated code

### LINKED ISSUE #1217: Invalid warning about exposed port using "12356/tcp"
## Problem description

When using `<port number="12345/tcp"/>` to explicitly expose a tcp port in a container I get this warning:

```
./kiwi/xml_parse.py:5076: UserWarning: Value "b'11211/tcp'" does not match xsd pattern restrictions: [['^(\\d+$|^\\d+/(udp$|^tcp))$']]
  warnings_.warn('Value "%s" does not match xsd pattern restrictions: %s' % (value.encode('utf-8'), self.validate_portnum_type_patterns_, ))
```
To me the pattern used for validation here looks somewhat fishy.

## Expected behaviour

Kiwi should accept  `<port number="12345/tcp"/>`

## OS and Software information
+ KIWI version:  9.18.13
+ Operating system: Tumbleweed
+ OBS version: -

### PR DESCRIPTION
**Pull request by davidcassany**
This commit updates xml_parser.py autogenerated code using a newer
generateDS version. The previous generateDS version is not properly
handling the regular expressions of some element's content.

Fixes #1217

### CODE REVIEW

File: kiwi/xml_parse.py [hunk 1]
```diff
@@ -5077,23 +6064,32 @@ def factory(*args_, **kwargs_):
...
         if value is not None and Validate_simpletypes_:
+            if not isinstance(value, str):
+                warnings_.warn('Value (%(value)s) is not of the correct base simple type (str)' % {"value": value, })
+                return
             if not self.gds_validate_simple_patterns(
                     self.validate_portnum_type_patterns_, value):
-                warnings_.warn('Value "%s" does not match xsd pattern restrictions: %s' % (value.encode('utf-8'), self.validate_portnum_type_patterns_, ))
-    validate_portnum_type_patterns_ = [['^(\\d+$|^\\d+/(udp$|^tcp))$']]
+                warnings_.warn('Value "%s" does not match xsd pattern restrictions: %s' % (encode_str_2_3(value), self.validate_portnum_type_patterns_, ))
+    validate_portnum_type_patterns_ = [['^((\\d+|\\d+/(udp|tcp)))$']]
```
Comment (davidcassany) [inline kiwi/xml_parse.py]: Here it can be seen that regular expressions are now properly handled.

**Comment by davidcassany** [comment]
@schaefi not much to review here :sweat_smile: But I thought about doing the PR to discuss if we bump generateDS version or not.

As far as I remember we added an specific version to build xml_parser.py because suddently started to appear annoying warnings when parsing the XML (warnings complaining about some content, however the config.xml was correct). In fact we added an specific version to avoid the issue reported in #1217 

Since this quite long ago and several versions of generateDS appeared meanwhile I thought about giving it a try to the latest version. I know this changes a sensitive part of the code that it is already proven to be functional. But I also doubt we should keep using old versions of generateDS for too long, we depend on this tool thus being close to upstream is important IMHO.

**Comment by davidcassany** [comment]
Ouch! just saw that is causes some tests to fail... :disappointed_relieved:

That's why I like the idea of trying to keep it up to date or at least time to time verify if the latest versions is good for us. This way we will not stress if someday the  version 2.29.14 disappears :P

Gonna have a look about to it tomorrow.

**Comment by schaefi** [comment]
would be great if we find a version bump of generateDS that brings it nearer to the latest and fixes the regexp problem.

I looked at the failed tests.

```
E   RecursionError: maximum recursion depth exceeded
```

caused by generated code but maybe this is because in ```delete_repository_sections_used_for_build``` the same repo returned by get_repository_sections() is used in self.xml_data.set_repository. I guess if we make a deep copy before set_repository that could solve it

**Comment by davidcassany** [comment]
> maybe this is because in `delete_repository_sections_used_for_build` the same repo returned by get_repository_sections() is used in self.xml_data.set_repository. I guess if we make a deep copy before set_repository that could solve it

Note the recursion happens before actually calling `set_repository`, happens within the comparison. I tested doing a deepcopy before the comparison is done and it causes a wrong effect, they are all considered to be different, however it breaks the recursion.

I filled a bug in generateDS here https://bitbucket.org/dkuhlman/generateds/issues/17/recursion-error-when-comparing-elements

**Comment by schaefi** [comment]
I think there are no open questions anymore correct ? Thanks for filing the bug at generateDS. For us we can now only find a version that works for us and at best also fixes the pattern bug. If not I suggest to the latest version that still works and wait for the fix on the bug. Does that makes sense ?

**Comment by davidcassany** [comment]
Correct no questions to answer here. From MY PoV I'd leave the PR opened or update the #1227 to state that we are waiting for a solution in generateDS code.

So I'd just wait for a new generateDS version that works for us. Meanwhile I have no strong opinion if we should better leave this PR opened or close it. Probably we could leave this opened for few days (this week) and see if the bug I filled in generateDS has some progression.

**Comment by schaefi** [comment]
The latest version that worked for me was 2.29.24, beginning with 2.30.8 the recursion problem was introduced. As far as I can see 2.29.24 also solves the pattern problem. Thus I added a commit here.
What do you think ?

File: kiwi/xml_parse.py [hunk 2]
```diff
@@ -5085,15 +5098,15 @@ def validate_portnum_type(self, value):
             if not self.gds_validate_simple_patterns(
                     self.validate_portnum_type_patterns_, value):
                 warnings_.warn('Value "%s" does not match xsd pattern restrictions: %s' % (value.encode('utf-8'), self.validate_portnum_type_patterns_, ))
-    validate_portnum_type_patterns_ = [['^(\\d+$|^\\d+/(udp$|^tcp))$']]
+    validate_portnum_type_patterns_ = [['^(\\d+|\\d+/(udp|tcp))$']]
```
[the remark this answers is not in the document]
Comment (schaefi) [reply kiwi/xml_parse.py]: @davidcassany please double check once tests are green. Thanks

[the remark this answers is not in the document]
Comment (davidcassany) [reply kiwi/xml_parse.py]: Correct this fixes the pattern :+1: Thanks for taking the time to review generateDS history

**Pull request merged**